### PR TITLE
feat: Use v5 of JS SDKs

### PR DESCRIPTION
**Pull request by HazAT**
### Tests

- [x] `1.7.16`
- [x] `1.8.8`
- [x] `2.0.10`
- [x] `3.0.2`
- [x] `4.2.10`
- [x] `5.0.10`
- [x] `6.0.7`

### CODE REVIEW

**Comment by claygriffiths** [comment]
Just chiming in to say we'd love to see this happen!

**Comment by HazAT** [comment]
@claygriffiths Cool :) 
So, I couldn't make the test run on electron 5/6 didn't spend enough time with it.
I tested it manually though and it seems to work just fine.
If you have time testing this branch in your app it would bring more assurance to ship this regardless.

**Comment by HazAT** [comment]
It's ready to review.
Test look ok'ish, some are failing on Windows we will investigate why.

I would love to have some app that already use the SDK to upgrade cc @timfish @claygriffiths

**Comment by timfish** [comment]
@HazAT how easy is it to get your build tooling publishing branches like this to some sort of "alpha" tag on npm? It would certainly help speed up testing.

Currently to test it with our apps, on all platforms we support I have to `git clone/checkout` this branch, `yarn build`, `yarn link`, `yarn link @sentry/electron` and then build our app. If it was published on npm I could update the dependency in single place and let our CI get on with all the work.

Using yarn link also doesn't resolve common dependencies that an app might have with `@sentry/electron` which means it's not strictly testing how it will function later when the dependency comes straight from npm.

**Comment by timfish** [comment]
Testing:
- [x] Windows
  - [x] JS
  - [x] Minidump
- [ ] macOS
  - [ ] JS
  - [ ] Minidump
- [ ] Linux
  - [ ] JS
  - [ ] Minidump

**Comment by timfish** [comment]
We are currently [doing this](https://github.com/getsentry/sentry-electron/issues/142#issuecomment-469286795) to get around the issues with Webpack bundling. 

Building our app works find on Windows but on macOS Catalina it fails when used via `yarn link` with an error:

```
    ERROR in /Users/timfish/Documents/Repositories/sentry-electron/dist/renderer/backend.js
    Module not found: Error: Can't resolve '@sentry/browser/dist/backend' in '/Users/timfish/Documents/Repositories/sentry-electron/dist/renderer'
     @ /Users/timfish/Documents/Repositories/sentry-electron/dist/renderer/backend.js 4:18-57
     @ /Users/timfish/Documents/Repositories/sentry-electron/dist/renderer/index.js
     @ ./src/error-reporting.ts
// snip
```

`@sentry/browser/dist/backend` is there if I follow normal node resolution paths but I've yet to work out if this is caused by `yarn link`, Webpack or maybe even [this](https://twitter.com/MarshallOfSound/status/1136526239139635200)... or a combination of all three!

**Comment by HazAT** [comment]
@timfish Good point, will release beta package.

**Comment by HazAT** [comment]
@timfish `1.0.0-beta.0` should be on npm also tagged with `next`.
cc @claygriffiths

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@
 There shouldn't be any breaking changes, interally we needed to do some changes because we were upgrading to use `5.x`
 of the JS/Node SDK.
 
+**Breaking**: We persist scope data to disk in case of a native crash. Since the internal scope data changed, existing
```
Comment (HazAT) [inline CHANGELOG.md]: cc @timfish

**Comment by claygriffiths** [comment]
Thanks all who were involved in implementing/testing this! We're looking forward to upgrading 😃

**Pull request merged**